components/ide/code/codehelper/main.go

```diff
@@ -258,3 +266,22 @@ func phaseLogging(phase string) context.CancelFunc {
 	}()
 	return cancel
 }
+
+func replaceOpenVSXUrl() error {
+	phase := phaseLogging("ReplaceOpenVSXUrl")
+	defer phase()
+	b, err := os.ReadFile(ProductJsonLocation)
+	if err != nil {
+		return errors.New("failed to read product.json: " + err.Error())
+	}
+	registryUrl := os.Getenv("VSX_REGISTRY_URL")
+	if registryUrl != "" {
+		b = bytes.ReplaceAll(b, []byte("https://open-vsx.org"), []byte(registryUrl))
+	}
+	b = bytes.ReplaceAll(b, []byte("{{extensionsGalleryItemUrl}}"), []byte("https://open-vsx.org/vscode/item"))
+	b = bytes.ReplaceAll(b, []byte("{{trustedDomain}}"), []byte("https://open-vsx.org"))
```

Comment:
@mustard-mh I have a dumb question, why this got added :sweat_smile:? I see they are added in dockerfile and then replaced here and in blobserve and they are just harcoded constant values :thinking:

Reply:
@akosyakov @iQQBot do you remember?

Reply:
Looks like can be removed.

Reply:
I know what happened, `https://open-vsx.org` is too normal to replace in `startup.sh`, so I need to replace it to something more special, in case it replaced by our openvsx-proxy.

Looks like we still need it?

Reply:
But the final value is harcoded here and in blobserve, where is openvsx-proxy updating this variable?

Reply:
It will replace them https://github.com/gitpod-io/gitpod/blob/6469258f2813001481e735b66ceaad1f9ad21f1f/components/ide/code/codehelper/main.go#L279

And installer will replace `https://open-vsx.org` again https://github.com/gitpod-io/gitpod/blob/6469258f2813001481e735b66ceaad1f9ad21f1f/install/installer/pkg/components/blobserve/configmap.go#L53

Reply:
yeah that's what I mean they are replaced with constant values, the same values that are already in product.json in gp-code/main branch, it doesn't change like `VSX_REGISTRY_URL` that changes for each SH installation.
So just remove them then?

Reply:
The constant value is `https://open-vsx.org` and `https://open-vsx.org/vscode/item` they all contains `open-vsx.org` which will be replaced in `installer` and `codehelper (startup.sh)`. 

But we don't want them to be changed.

- In `codehelper`, we can use [sjson](https://github.com/tidwall/sjson) to Set value with json path we want
- In `installer`, it's just global replacement

So we still need them because of installer

Reply:
Ah I get it now :bulb: , but then can we make the search more specific (with a regex maybe) so it doesn't replace the other values, right now it's not obvious at all